lbrynet/schema/attrs.py

```diff
@@ -204,24 +204,28 @@ class Fee(Metadata):
     __slots__ = ()
 
     def update(self, address: str = None, currency: str = None, amount=None):
+        if amount and not (currency or self.message.currency):
+            raise Exception('In order to set a fee amount, please specify a fee currency')
+        if currency and not (amount or self.message.amount):
+            raise Exception('In order to set a fee currency, please specify a fee amount')
         if address is not None:
             self.address = address
-        if currency is not None and amount is not None:
-            currency = currency.lower()
+        if currency or amount:
+            currency = currency.lower() if currency is not None else self.currency.lower()
             assert currency in ('lbc', 'btc', 'usd'), f'Unknown currency type: {currency}'
-            setattr(self, currency, Decimal(amount))
+            setattr(self, currency, Decimal(amount or self.amount))
```

Comment:
I think this can be cleaned up more. You check `currency` multiple times, including the inline. Are there really that many different cases? Maybe just check the good case and do `setattr` if everything needed is there, then have an `else` that handles the bad case. I think that might result in clearer code.